src/subspace_store.c

```diff
@@ -104,50 +105,79 @@ subspace_store_add(SubspaceStore *store, const Hypercube *hc,
 
 		if (node == NULL)
 		{
+			/*
+			 * We should have one internal node per dimension in the
+			 * hypertable. If we don't have one for the current dimension,
+			 * create one now. (There will always be one for time)
+			 */
 			Assert(last != NULL);
-			last->storage = subspace_store_internal_node_create(i == hc->num_slices - 1);
+			last->storage = subspace_store_internal_node_create(i == (hc->num_slices - 1));
 			last->storage_free = subspace_store_internal_node_free;
 			node = last->storage;
 		}
 
+		Assert(store->max_items == 0 || node->descendants <= (size_t)store->max_items);
```

Comment:
Hm, do you necessarily need this assert? It looks identical to the one you added at the end of the function, and there is an assert inside the `if(over max_items)` segment that should catch anything like this.

Reply:
Probably not strictly necessary, but it only fires during debug builds, is easier to read, and `Assert(store->max_items + 1 == node->descendants);` has some different failures in the case of overflow.